java/native/src/CMakeLists.txt

```diff
@@ -0,0 +1,49 @@
+cmake_minimum_required(VERSION 2.8.0)
+
+set(ignoreMe "${SBT}") # sbt-jni defines -DSBT
+set(MAKE_COLOR_MAKEFILE ON)
+
+# Define project and related variables
+# (required by sbt-jni) please use semantic versioning
+#
+project (gdaljni)
+set(PROJECT_VERSION_MAJOR 0)
+set(PROJECT_VERSION_MINOR 0)
+set(PROJECT_VERSION_PATCH 1)
+
+if (APPLE)
+  set(CMAKE_MACOSX_RPATH ON)
+  SET(CMAKE_SKIP_BUILD_RPATH TRUE)
+  SET(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
+  SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+endif ()
+
+# Setup JNI
+find_package(JNI REQUIRED)
+if (JNI_FOUND)
+    message (STATUS "JNI include directories: ${JNI_INCLUDE_DIRS}")
+endif()
+
+# Depend on GDAL
+find_package(GDAL REQUIRED)
+
+# !IMPORTANT! We also need to add here your files from the ../src folder
```

Comment:
Also an example `CMakeLists` file. I hope it would be useful. Also you could notice that I'm using a cpp compiler, so you can use whatever you think is appropriate.